lib/Driver/Job.cpp

```diff
@@ -52,6 +52,18 @@ void CommandOutput::ensureEntry(StringRef PrimaryInputFile,
   }
 }
 
+void CommandOutput::addOutputs(CommandOutput const &other) {
+  assert(PrimaryOutputType == other.PrimaryOutputType);
+  assert(&DerivedOutputMap == &other.DerivedOutputMap);
+  Inputs.append(other.Inputs.begin(),
+                other.Inputs.end());
+  types::forAllTypes([&](types::ID Type) {
```

Comment:
SmallSet has an iterator-based `insert` which will be equivalent.

Reply:
Yes, as a sink; but it doesn't have an accessor to give you iterator(s) as a source!

Reply:
Oh, bah! Okay then.

Reply:
That said, if you don't care about adding to an *existing* map, you could just do a copy-assignment.

Reply:
In this case we're doing a merge, the moral equivalent of +=, so I do care about the existing entries (if I'm understanding your comment correctly?)

Reply:
I understand that the operation is written as `+=`, but at the only place you use it it'll be empty or identical anyway, right?

Reply:
Oh, fair point. I'll modify it to assert that.